src/mlpack/methods/ann/init_rules/glorot_init.hpp

```diff
@@ -153,8 +160,9 @@ inline void GlorotInitializationType<true>::Initialize(arma::Mat<eT>& W,
 }
 
 template<>
-template<typename eT>
-inline void GlorotInitializationType<true>::Initialize(arma::Mat<eT>& W)
+template<typename MatType,
+     typename std::enable_if_t<IsMat<MatType>::value, bool>>
```

Comment:
If you take the change from the declaration (using `typename = typename std::enable_if_t<...>` instead), then the second template parameter is just an unnamed parameter, so you could do 

```
template<typename MatType, typename>
```

here.